extensions/warp-pd-flatfile/src/lib.rs

```diff
@@ -361,6 +363,8 @@ impl PocketDimension for FlatfileStorage {
 
                         let new_path = {
                             let mut path = self.directory.clone();
+                            path.push("FileSystem");
+                            create_dir_all(&path)?;
```

Comment:
What you might want to do is perform a check to determine of the directory exists and if not to create the directory. I believe `std::fs::create_dir_all` will error when there is an existing directory